### DISCUSSION TITLE
TypeError: url_1.URL is not a constructor at Telegram.callApi

### DISCUSSION THREAD

**Discussion by hieupcvn** [telegraf/telegraf#1848] [Q&A]
Hello, could you please assist me.
const bot = new Telegraf(MY_TOKEN);
bot.start((ctx) => ctx.reply("Welcome"));
bot.launch(); ====> TypeError: url_1.URL is not a constructor at Telegram.callApi

**Comment by MKRhere** [comment]
Node version?

**Reply by hieupcvn** [reply]
Hi MKRhere, I am using node version v16.19.0

**Reply by MKRhere** [reply]
Should be fine in Node 16. Are you trying to bundle telegraf?

**Reply by hieupcvn** [reply]
yes

**Reply by MKRhere** [reply]
If you're trying to run Telegraf in the browser, this is currently not officially supported, but other users have had luck making it work. You'll want to instruct Webpack/Rollup/your bundling tool to replace the URL import and use global URL instead. I'll look into making this is the default in the next minor version.

In the future, Telegraf v5 will officially support browsers natively.

**Comment by Ovler-Young** [comment]
I'm using telegraf in Cloudflare Workers and had the same problem. After some search and lots of test, I found that add this line in index.js or import this line will resolve this problem.
```
require('url').URL = URL;
```
The code is taken from the migration from v3 to v4 of [cfworker-middleware-telegraf](https://github.com/Tsuk1ko/cfworker-middleware-telegraf/commit/e745e7652d2c0967fb7a4b65c2c49b59a428543a#diff-d477fce6616e0599e8db47a0bac6fdaba8934480f6f3a5400c992895c433a949)
